### ISSUE TITLE
Build immediately fails on OS X with Jetbrains Rider IDE

### ISSUE THREAD

**Issue by jeffreyschultz** [spring-net-futures/spring-net#1]
The Spring.Core.2010 project has a build step that depends on executing antlr.exe to generate parser code from the *.g files. Obviously, this limits the ability to reproducibly build the project on other platforms.

The IDE shouldn't matter, but I have been using Rider for development at work for a while now, and have encountered minimal issues.

```
  Microsoft.Common.CurrentVersion.targets(1290, 5): [MSB3073] The command "rem /Users/jschultz/Source/spring-net-futures/spring-net/src/Spring/Spring.Core/../../../build-support/tools/antlr-2.7.6/antlr-2.7.6.exe -o /Users/jschultz/Source/spring-net-futures/spring-net/src/Spring/Spring.Core/Expressions/Parser /Users/jschultz/Source/spring-net-futures/spring-net/src/Spring/Spring.Core/Expressions/Expression.g" exited with code 127.
```

**Comment by jeffreyschultz** [comment]
I am looking into what it would take to upgrade the antlr used to antlr 4.7, which as of version 4.4, now includes the mono-compatible C# target together with the main tool. There was another separate tool maintained for supporting C#, but that has now been deprecated.

https://github.com/antlr/antlr4/tree/master/runtime/CSharp

**Comment by jeffreyschultz** [comment]
I realized that the pre-build step was actually commented out, and since I am working on OS X, the shell didn't understand the _rem_ statement. I commented out the pre-build step and it compiled. I realized that build step is only needed to update the generated parser code when the grammar is updated, and is not executed during every build.

I do believe that going forward antlr does need to be updated to make this more cross-platform compatible.
